commit 5cb579318a18d280cb43e0aa6039ab7f4d760c45
Author: bertin <macportsuser@macports.org>
Date:   Fri May 23 21:30:17 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts-pextlib/files/lamarck/patch-xinstall-have_setmode.diff

diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index 6d826d71..148e54cc 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -119,6 +119,8 @@ extern char **environ;
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 __attribute__((format(printf, 3, 0)))
diff --git a/src/pextlib1.0/xinstall.c b/src/pextlib1.0/xinstall.c
index 2199a229..a1863a48 100644
--- a/src/pextlib1.0/xinstall.c
+++ b/src/pextlib1.0/xinstall.c
@@ -116,6 +116,8 @@ extern int copyfile(const char *from, const char *to, void *state,
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 /* Bootstrap aid - this doesn't exist in most older releases */
